### ISSUE TITLE
A5 Q2 Part c Using Git Push Command and declaring no arguments 

### ISSUE THREAD

**Issue by maryammahjoob** [BME1478H/Fall2020class#329]
Hello, 

For Assignment 5, question 2, Part c, I am a bit confused between the difference of using `git push ` versus `git push origin`. I used `git push`, and it appears the changes have been recorded on my local repository on github. Are either okay to use? 

Thank you in advance!

**Comment by constructor-s** [comment]
According to [Git documentation](https://git-scm.com/docs/git-push):

> When the command line does not specify where to push with the <repository> argument, branch.*.remote configuration for the current branch is consulted to determine where to push. If the configuration is missing, it defaults to origin.
> 
> When the command line does not specify what to push with <refspec>... arguments or --all, --mirror, --tags options, the command finds the default <refspec> by consulting remote.*.push configuration, and if it is not found, honors push.default configuration to decide what to push (See git-config[1] for the meaning of push.default).
> 
> When neither the command-line nor the configuration specify what to push, the default behavior is used, which corresponds to the simple value for push.default: the current branch is pushed to the corresponding upstream branch, but as a safety measure, the push is aborted if the upstream branch does not have the same name as the local one.

TL;DR: There are some default values at work both with respect to Git defaults and your repository defaults, such that for typical use cases your two commands are equivalent.

**Comment by maryammahjoob** [comment]
> According to [Git documentation](https://git-scm.com/docs/git-push):
> 
> > When the command line does not specify where to push with the  argument, branch.*.remote configuration for the current branch is consulted to determine where to push. If the configuration is missing, it defaults to origin.
> > When the command line does not specify what to push with ... arguments or --all, --mirror, --tags options, the command finds the default  by consulting remote.*.push configuration, and if it is not found, honors push.default configuration to decide what to push (See git-config[1] for the meaning of push.default).
> > When neither the command-line nor the configuration specify what to push, the default behavior is used, which corresponds to the simple value for push.default: the current branch is pushed to the corresponding upstream branch, but as a safety measure, the push is aborted if the upstream branch does not have the same name as the local one.
> 
> TL;DR: There are some default values at work both with respect to Git defaults and your repository defaults, such that for typical use cases your two commands are equivalent.

Thank you I appreciate this answer it is much more clear now!
